sql/expression/function/greatest_least.go

```diff
@@ -0,0 +1,329 @@
+package function
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"gopkg.in/src-d/go-errors.v1"
+	"gopkg.in/src-d/go-mysql-server.v0/sql"
+)
+
+// Greatest returns the argument with the greatest numerical or string value. It allows for
+// numeric (ints anf floats) and string arguments and will return the used type
+// when all arguments are of the same type or floats if there are numerically
+// convertible strings or integers mixed with floats. When ints or floats
+// are mixed with non numerically convertible strings, those are ignored.
+type Greatest struct {
+	args []sql.Expression
+}
+
+// ErrUnsupportedType is returned when an argument to Greatest or Latest is not numeric or string
+var ErrUnsupportedType = errors.NewKind("unsupported type for greatest/latest argument: %T")
+
+// NewGreatest creates a new Greatest UDF
+func NewGreatest(args ...sql.Expression) (sql.Expression, error) {
+	if len(args) == 0 {
+		return nil, sql.ErrInvalidArgumentNumber.New("GREATEST", "1 or more", 0)
+	}
+
+	for _, arg := range args {
+		if !arg.Resolved() {
+			continue
+		}
+
+		if sql.IsTuple(arg.Type()) {
+			return nil, sql.ErrInvalidType.New("tuple")
+		}
+	}
+
+	return &Greatest{args}, nil
+}
+
+// Type implements the Expression interface.
+func (f *Greatest) Type() sql.Type { return sql.Float64 }
```

Comment:
float? so what about strings?

Reply:
float?

Reply:
Looks like I left this when I started to develop this just for floats. But it works :-? Is there any way I can specify several types there? (this could return float64, int64 or string depending on the arguments).

Reply:
Take a look at the Comparison type logic to determine which type to use.

Reply:
`sql/expression/comparison.go::Comparison` seems to return a `Boolean`. I updated the PR adding a type member and updating it after `Eval`, but I'm not sure that's right, could `Type()` be called before `Eval()`?

Reply:
Yup, type can and will be called before Eval.

I meant this logic: https://github.com/src-d/gitbase/issues/808

Reply:
Implemented, thanks for the tip.